src/components/PostsList/PostsList.jsx

```diff
@@ -1,37 +1,56 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import './PostsList.scss';
 
-export const PostsList = () => (
+export const PostsList = ({ posts, postId, onPostIdSelect, onPostClose }) => (
   <div className="PostsList">
     <h2>Posts:</h2>
 
     <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
+      {posts.map(post => (
+        <li
+          key={post.id}
+          className="PostsList__item"
         >
-          Close
-        </button>
-      </li>
+          <div>
+            <b>
+              {`[User #${post.userId}]:`}
+            </b>
+            {post.title}
+          </div>
 
-      <li className="PostsList__item">
-        <div>
-          <b>[User #2]: </b>
-          et ea vero quia laudantium autem
-        </div>
+          {post.id !== postId && (
+            <button
+              type="button"
+              className="PostsList__button button"
+              onClick={() => onPostIdSelect(post.id)}
+            >
+              Open
+            </button>
+          )}
```

Comment:
нет необходимости по отдельности прописывать кнопки, будто это разные кнопки. Можно записать все так:
```suggestion
          <button
            type="button"
            className="PostsList__button button"
            onClick={() => {
              if (post.id !== postId) {
                onPostIdSelect(post.id);
              } else {
                onPostClose();
              }
            }}
          >
            {(post.id !== postId) ? 'Open' : 'Close'}
          </button>
```